docs/edgeql/paths.rst

```diff
@@ -36,33 +41,87 @@ Consider the following schema:
       required author: User;
     }
 
+    type Comment {
+      required property text: str;
+      required link author: User;
+    }
+
+A few simple inserts will allow some experimentation with paths.
+
+Start with a first user:
+
+.. code-block:: edgeql-repl
+
+    db> insert User {
+    ... email := "user1@me.com",
+    ... };
+
+Along comes another user who adds the first user as a friend:
+
+.. code-block:: edgeql-repl
+
+    db> insert User {
+    ... email := "user2@me.com",
+    ... friends := (select detached User filter .email = "user1@me.com")
+    ... };
+
+The first user reciprocates, adding the new user as a friend:
+
+.. code-block:: edgeql-repl
+
+    db> update User filter .email = "user1@me.com" 
+    ... set { 
+    ... friends += (select detached User filter .email = "user2@me.com")
+    ... };
+
+The second user writes a blog post about how nice EdgeDB is:
+
+.. code-block:: edgeql-repl
+
+    db> insert BlogPost {
+    ... title := "EdgeDB is awesome",
+    ... author := assert_single((select User filter .email = "user2@me.com"))
+    ... };
+
+And the first user follows it up with a comment below the post:
+
+.. code-block:: edgeql-repl
+
+    db> insert Comment {
+    ... text := "Nice post, user2!",
+    ... author := assert_single((select User filter .email = "user1@me.com"))
+    ... };
+
 The simplest path is simply ``User``. This is a :ref:`set reference
 <ref_eql_set_references>` that refers to all ``User`` objects in the database.
 
 .. code-block:: edgeql
 
-  select User;
+    select User;
 
 Paths can traverse links. The path below refers to *all Users who are the
 friend of another User*.
 
 .. code-block:: edgeql
 
-  select User.friends;
+    select User.friends;
 
 Paths can traverse to an arbitrary depth in a series of nested links.
+The ``select`` below ends up showing the author of the BlogPost
+(The friend of the friend of the BlogPost, which is the author)
 
 .. code-block:: edgeql
 
-  select BlogPost.author.friends.friends;
+    select BlogPost.author; # The author
+    select BlogPost.author.friends.friends; # The author again
```

Comment:
I like the wording you suggested here because it describes accurately what it does but is also enough of a tongue twister that the reader will know that this a demonstration and certainly not a recommended practice or anything.

https://github.com/edgedb/edgedb/commit/7f10efeb9fab8e6047073e2756c7ff296f58cb4c